Compilation error with Micronaut Data 2.3.1-SNAPSHOT and Micronaut 2.4.0-SNAPSHOT

The GraalVM CI has detected a compilation error with all the JDBC/JPA test applications. It happens when using the latest snapshots for Micronaut and Micronaut Data.

```
> Task :compileJava FAILED
/home/ivan/workspaces/micronaut-graal-tests/micronaut-data-jdbc-graal/src/main/java/example/domain/Pet.java:24: warning: Usages of deprecated annotation javax.annotation.Nullable found. You should use io.micronaut.core.annotation.Nullable instead.
    public Pet(String name, @Nullable Owner owner) {
                                            ^
Note: Writing reflect-config.json file to destination: META-INF/native-image/example.jdbc/micronaut-data-jdbc-graal/reflect-config.json
Note: Writing resource-config.json file to destination: META-INF/native-image/example.jdbc/micronaut-data-jdbc-graal/resource-config.json
Note: Creating bean classes for 7 type elements

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> java.lang.NoSuchMethodError: io.micronaut.inject.processing.JavaModelUtils.getTypeReference(Lio/micronaut/inject/ast/TypedElement;)Lorg/objectweb/asm/Type;
```

### Steps to Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-data-jdbc-graal`  
- `cd micronaut-data-jdbc-graal`
- `git checkout 2.4.x_h2`
- `./gradlew run`


### Expected Behaviour

The application should compile

### Actual Behaviour

The application doesn't compile. See previous error.

### Environment Information

- **Operating System**: Linux Mint 20.1
- **Micronaut Version:** 2.4.0-SNAPSHOT and Micronaut Data 2.3.1-SNAPSHOT
- **JDK Version:** 8

--- a/inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java
+++ b/inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java
@@ -45,8 +45,6 @@ import io.micronaut.inject.writer.BeanDefinitionReferenceWriter;
 import io.micronaut.inject.writer.BeanDefinitionVisitor;
 import io.micronaut.inject.writer.BeanDefinitionWriter;
 import io.micronaut.inject.writer.OriginatingElements;
-import org.objectweb.asm.Type;
-
 import javax.annotation.processing.ProcessingEnvironment;
 import javax.annotation.processing.RoundEnvironment;
 import javax.annotation.processing.SupportedOptions;
@@ -1300,8 +1298,7 @@ public class BeanDefinitionInjectProcessor extends AbstractInjectAnnotationProce
                                 for (int i = 0; i < adaptedArgumentTypes.length; i++) {
                                     ParameterElement parameterElement = sourceParams[i];
                                     final ClassElement genericType = parameterElement.getGenericType();
-                                    final Type t = JavaModelUtils.getTypeReference(genericType);
-                                    adaptedArgumentTypes[i] = new AnnotationClassValue<>(t.getClassName());
+                                    adaptedArgumentTypes[i] = new AnnotationClassValue<>(JavaModelUtils.getClassname(genericType));
                                 }
 
                                 MethodElement javaMethodElement = elementFactory.newMethodElement(

--- a/inject/src/main/java/io/micronaut/inject/processing/JavaModelUtils.java
+++ b/inject/src/main/java/io/micronaut/inject/processing/JavaModelUtils.java
@@ -253,4 +253,14 @@ public class JavaModelUtils {
             }
         }
     }
+
+    /**
+     * Return the type reference for a class.
+     *
+     * @param type The type
+     * @return The {@link Type}
+     */
+    public static String getClassname(TypedElement type) {
+        return getTypeReference(type).getClassName();
+    }
 }

